designer/fitting/dwipi.py

```diff
@@ -1655,175 +1656,294 @@ def optimPipeline(self, savePath=os.getcwd()):
+            raise Exception('Threshold cannot be less than 0 or greater '
+                            'than 1. Please specifty a between the range '
+                            '[0 1].')
+        if brainmask is None:
+            self.BrainMask = np.ones((self.Img.shape[0], self.Img.shape[
+                1], self.Img.shape[2]), dtype=bool, order='F')
+        else:
+            self.BrainMask = np.array(brainmask).astype(bool)
+        if tissueth is not None:
+            self.TissueMask = self.Img > self.TissueThreshold
+            self.BrainMask = np.logical_and(
+                self.BrainMask, self.TissueMask)
+        else:
+            self.TissueMask = self.BrainMask
         # Get box filter properties
-        centralIdx = np.median(range(self.Size))
-        d2move = np.int(np.abs(self.Size - (centralIdx + 1)))  # Add 1 to central idx because first index starts with zero
-
-        violIdx = np.array(np.where(self.Mask))   # Locate coordinates of violations
-        nvox = violIdx.shape[1]
-        self.PatchIdx = np.zeros(violIdx.shape[1])
-
-        inputs = tqdm(range(nvox))
-        cntSkip = 0
+        centralIdx = np.median(range(self.Size)).astype(int)
+        # Add 1 to central idx because first index starts with zero
+        d2move = np.int(np.abs(self.Size - (centralIdx + 1)))
+        # Apply a nan padding to all 3 dimensions of the input image and a
+        # nan padding to mask. Padding widths is same distance between
+        # centroid of patch to edge. This enables median filtering of edges
+        self.Img = np.pad(self.Img, d2move, 'constant',
+                          constant_values=np.nan)
+        self.BrainMask = np.pad(self.BrainMask, d2move, 'constant',
+                                constant_values=np.nan)
+        # Create (3 x nvox) cartesian coordinate representation of image
+        # within a brainmask
+        cartIdx = np.array(np.where(
+            np.multiply(self.Img, self.BrainMask) > 0))
+        nvox = cartIdx.shape[1]
+        inputs = tqdm(range(nvox),
+                      desc='Median Outlier Mask     ',
+                      unit='vox',
+                      ncols=tqdmWidth)
+        self.OutlierMask = np.zeros(self.BrainMask.shape, dtype=bool)
         for i in inputs:
             # Index beginning and ending of patch
-            Ib = violIdx[0, i] - d2move
-            Ie = violIdx[0, i] + d2move + 1     # Mitigate Python's [X,Y) indexing
-            Jb = violIdx[1, i] - d2move
-            Je = violIdx[1, i] + d2move + 1     # Mitigate Python's [X,Y) indexing
-            Kb = violIdx[2, i] - d2move
-            Ke = violIdx[2, i] + d2move + 1     # Mitigate Python's [X,Y) indexing
-
+            Ib = cartIdx[0, i] - d2move
+            Ie = cartIdx[0, i] + d2move
+            Jb = cartIdx[1, i] - d2move
+            Je = cartIdx[1, i] + d2move
+            Kb = cartIdx[2, i] - d2move
+            Ke = cartIdx[2, i] + d2move
```

Comment:
Not a requested change, but why did you get rid of the +1? Was the original indexing incorrect?

Reply:
I accounted for it by adding +1 where these indexes are actually used.